### ISSUE TITLE
After accepting finishing merge clusters, main link to session goes to merge entities not final match

### ISSUE THREAD

**Issue by fgregg** [datamade/dedupe-api#175]
```
INFO:werkzeug:127.0.0.1 - - [02/Apr/2015 20:25:22] "GET /get-canon-review-cluster/?session_id=85d2d365-9c6d-4721-9221-1422cfea9fda HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [02/Apr/2015 20:25:29] "GET /mark-all-canon-clusters/?session_id=85d2d365-9c6d-4721-9221-1422cfea9fda&action=yes HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [02/Apr/2015 20:25:29] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [02/Apr/2015 20:25:30] "GET /session-list/ HTTP/1.1" 200 -
...
INFO:werkzeug:127.0.0.1 - - [02/Apr/2015 20:29:17] "GET /session-list/ HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [02/Apr/2015 20:29:21] "GET /session-review/?session_id=85d2d365-9c6d-4721-9221-1422cfea9fda&second_review=True HTTP/1.1" 200 -
ERROR:sqlalchemy.pool.QueuePool:Error closing cursor
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/engine/base.py", line 951, in _execute_context
    context)
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/engine/default.py", line 436, in do_execute
    cursor.execute(statement, parameters)
psycopg2.ProgrammingError: relation "cr_85d2d365-9c6d-4721-9221-1422cfea9fda" does not exist
LINE 6:             FROM "cr_85d2d365-9c6d-4721-9221-1422cfea9fda" A...
                         ^


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/sqlalchemy/engine/base.py", line 1057, in _safe_close_cursor
    cursor.close()
psycopg2.InternalError: current transaction is aborted, commands ignored until end of transaction block

INFO:werkzeug:127.0.0.1 - - [02/Apr/2015 20:29:25] "GET /get-canon-review-cluster/?session_id=85d2d365-9c6d-4721-9221-1422cfea9fda HTTP/1.1" 500 -
INFO:werkzeug:127.0.0.1 - - [02/Apr/2015 20:29:29] "GET /session-list/ HTTP/1.1" 200 -
...
```

**Comment by fgregg** [comment]
I think this is because we don't move the session until https://github.com/datamade/dedupe-api/blob/master/api/utils/delayed_tasks.py#L247 

Shouldn't we advance the session to the next state as soon as we are done merging entities?

**Comment by evz** [comment]
So, you're talking about when you manually work down the whole queue instead of bulk accepting?

**Comment by fgregg** [comment]
Hmm.. I don't quite recall the mode.

**Comment by evz** [comment]
I think I fixed it here: e10c7b783d9cb2e9c4804d3e9e151cd1f9ee2e42
